src/main/host/descriptor/socket/inet/udp.rs

```diff
@@ -0,0 +1,892 @@
+use std::io::Read;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+use nix::errno::Errno;
+use nix::sys::socket::{AddressFamily, MsgFlags, Shutdown, SockaddrIn};
+use shadow_shim_helper_rs::syscall_types::ForeignPtr;
+
+use crate::core::worker::Worker;
+use crate::cshadow as c;
+use crate::host::descriptor::socket::inet::{self, InetSocket};
+use crate::host::descriptor::socket::{RecvmsgArgs, RecvmsgReturn, SendmsgArgs, ShutdownFlags};
+use crate::host::descriptor::{
+    File, FileMode, FileState, FileStatus, OpenFile, Socket, StateEventSource, StateListenerFilter,
+    SyscallResult,
+};
+use crate::host::memory_manager::MemoryManager;
+use crate::host::syscall::io::{write_partial, IoVec, IoVecReader};
+use crate::host::syscall_types::SyscallError;
+use crate::network::net_namespace::NetworkNamespace;
+use crate::network::packet::{PacketRc, PacketStatus};
+use crate::utility::callback_queue::{CallbackQueue, Handle};
+use crate::utility::sockaddr::SockaddrStorage;
+use crate::utility::{HostTreePointer, ObjectCounter};
+
+/// Maximum size of a datagram we are allowed to send out over the network.
+const CONFIG_DATAGRAM_MAX_SIZE: usize = 65507;
```

Comment:
Where did this particular number come from? (Might be worth a comment)

Reply:
Added a comment.

> 65,535 (2^16 - 1) - 20 (ip header) - 8 (udp header)